### ISSUE TITLE
Multiple Quill Editor in a ListView result in incorrect AutoScroll behavior when text goes behind soft keyboard

### ISSUE THREAD

**Issue by Mustaqode** [singerdmx/flutter-quill#2404]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/singerdmx/flutter-quill/issues)

### Flutter Quill version

10.8.5

### Steps to reproduce

1. Add multiple Quill Editors inside a listview (dynamic addition if possible)
2. Use the common _scrollController for the listview and the Quill Editors
3. Now type in any one of the quill editors; when it reaches near the keyboard you can see a weird random scroll behavior. 

### Expected results

The scroll should be just fine like we use the default TextField!

### Actual results

The screen scrolls weirdly so that the current cursor goes out of view!!!

### Additional Context

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

**Comment by Mustaqode** [comment]
From a discussion in slack =>

> Ok let me rephrase it, we have a scrollcontroller for the listview, and we have quill as a part of the listview. We might have 1-10 editors in the screen added dynamically. Quill doesn’t support the basic scroll functionality that normal TextFields have in default, which is scrolling the screen up if the text is obscured by the keyboard; so we need a workaround that uses the main scrollController of the listview with respect to caret position of the editor in focus.  So that if the cursor hides behind the keyboard we scroll the list up. Is there any possible workarounds?
> Or it will be great if someone fixes it in Quill itself!

**Comment by manudicri** [comment]
Same problem! Any news about this?
Thank you

**Comment by shovelmn12** [comment]
@Mustaqode 

Same issue using slivers !! I also tried going back to `Column` and I got the same result
Swapping with `TextField` fixes the issue, my best guess is that the implementation of `QuillEditor` does not follow the same behaviour as `TextField` therefore it's not playing nicely with flutter.

**Comment by manudicri** [comment]
@shovelmn12 
This is what fixed it for me. But I don't know if your problem is different.
https://github.com/singerdmx/flutter-quill/issues/1599#issuecomment-2782655568

**Comment by shovelmn12** [comment]
@manudicri I just tried it, does not work with slivers :(
